com.unity.multiplayer.mlapi/Runtime/Profiling/PerformanceTickData.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace MLAPI.Profiling
+{
+    public class PerformanceTickData
+    {
+        public int tickID;
+
+        readonly ProfilingDataStore m_TickData = new ProfilingDataStore();
+
+        public void Increment(string fieldName, int count = 1)
+        {
+            m_TickData.Increment(fieldName, count);
+        }
+
+        public void AddNonDuplicateData(IReadOnlyDictionary<string, int> transportProfilerData)
+        {
+            IEnumerable<KeyValuePair<string, int>> nonDuplicates = transportProfilerData.Where(entry => !m_TickData.HasData(entry.Key));
```

Comment:
this linq query is unnecessary here. you could do the HasData check inside the loop and avoid the linq cost

Reply:
https://stackoverflow.com/questions/4044400/linq-performance-faq

The allocations and performance AFAIK are the same. `nonDuplicates` isn't an allocation and we probably save speed by iterating over less entries.
That is to say, the line in question doesn't get evaluated until the bottom `for loop` lazily. And I feel that this reads better.

Reply:
Almost all Linq queries, including `.Where`, allocate an IEnumerable on the heap to wrap the iteration of the collection. this line also allocates an anonymous function - it can't be static since it captures the member field `m_TickData`

I won't hold approval for this since it's not a high frequency alloc but I do think it could be optimized